core/src/main/java/org/apache/brooklyn/core/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -417,6 +419,53 @@ else for (String key: otherKeys) {
         return result;
     }
 
+    public static Map<?,?> getCatalogMetadata(String yaml) {
+        Map<?,?> itemDef = Yamls.getAs(Yamls.parseAll(yaml), Map.class);
+        return getFirstAsMap(itemDef, "brooklyn.catalog").orNull();        
+    }
+    
+    public static VersionedName getVersionedName(Map<?,?> catalogMetadata) {
+        String id = getFirstAs(catalogMetadata, String.class, "id").orNull();
+        String version = getFirstAs(catalogMetadata, String.class, "version").orNull();
+        String symbolicName = getFirstAs(catalogMetadata, String.class, "symbolicName").orNull();
+        symbolicName = findAssertingConsistentIfSet("symbolicName", symbolicName, getFirstAs(catalogMetadata, String.class, "symbolic-name").orNull());
+        // prefer symbolic name and version, if supplied
+        // else use id as symbolic name : version or just symbolic name
+        // (must match if both supplied)
+        if (Strings.isNonBlank(id)) {
+            if (CatalogUtils.looksLikeVersionedId(id)) {
+                symbolicName = findAssertingConsistentIfSet("symbolicName", symbolicName, CatalogUtils.getSymbolicNameFromVersionedId(id));
```

Comment:
Not sure I agree with this (and I think it contradicts what was discussed on the mailing list). Do we want to support something like the (contrived) example below:
```
brooklyn.catalog:
  symbolicName: com.example.brooklyn.test-utils
  id: test-case
  version: 1.0
  item:
    type: org.apache.brooklyn.test.framework.TestCase
    ...
```
Here, the name-space for the entity "test-case" is entirely different from the namespace of the bundle ("com.example..."). Your code would force that the symbolicName and id are identical.